src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -741,9 +742,12 @@ def save_as_table(
         clustering_keys: Iterable[str],
         comment: Optional[str],
         child: SnowflakePlan,
+        source_plan: Optional[LogicalPlan],
+        use_scoped_temp_objects: bool,
+        is_generated: bool,  # true if the table is generated internally
     ) -> SnowflakePlan:
         full_table_name = ".".join(table_name)
```

Comment:
can we add a check at the very beginning that if is_generated is true, the SaveMode today must be ERROR_IF_EXISTS. Otherwise it is kind of wired that why other mode doesn't support is_generated.

Reply:
and also add a test that when using other mode with is_generated, it errors out today

Reply:
I was wondering if we make is_generated compatible with ERROR_IF_EXISTS and APPEND. is_generated + APPEND will be used to run create and insert plan, and is_generated + ERROR_IF_EXISTS will be used to create CTAS plan